Library/Homebrew/extend/ENV/super.rb

```diff
@@ -67,6 +67,11 @@ def setup_build_environment(formula = nil)
     self["HOMEBREW_INCLUDE_PATHS"] = determine_include_paths
     self["HOMEBREW_LIBRARY_PATHS"] = determine_library_paths
 
+    if MacOS::Xcode.without_clt?
+      self["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version.to_s
+      self["SDKROOT"] = MacOS.sdk_path
```

Comment:
That's no longer true though, “thanks” to one of my other suggestions: If the OS X is too new for the active Xcode, this will now return `nil`.